File: python/svix/internal/openapi_client/models/background_task_type.py
from enum import Enum


class BackgroundTaskType(str, Enum):
    APPLICATION_STATS = "application.stats"
    ENDPOINT_RECOVER = "endpoint.recover"
    ENDPOINT_REPLAY = "endpoint.replay"
    EVENT_TYPE_AGGREGATE = "event-type.aggregate"
    MESSAGE_BROADCAST = "message.broadcast"
    SDK_GENERATE = "sdk.generate"

    def __str__(self) -> str:
        return str(self.value)

File: java/lib/src/main/java/com/svix/models/BackgroundTaskType.java
// This file is @generated
package com.svix.models;

import com.fasterxml.jackson.annotation.JsonValue;
import com.svix.Utils.ToQueryParam;

public enum BackgroundTaskType implements ToQueryParam {
    ENDPOINT_REPLAY("endpoint.replay"),
    ENDPOINT_RECOVER("endpoint.recover"),
    APPLICATION_STATS("application.stats"),
    MESSAGE_BROADCAST("message.broadcast"),
    SDK_GENERATE("sdk.generate"),
    EVENT_TYPE_AGGREGATE("event-type.aggregate"),
    APPLICATION_PURGE_CONTENT("application.purge_content"),
    ENDPOINT_BULK_REPLAY("endpoint.bulk_replay");
    private final String value;

    BackgroundTaskType(String value) {
        this.value = value;
    }

    @JsonValue
    public String getValue() {
        return this.value;
    }

    @Override
    public String toQueryParam() {
        return this.value;
    }
}
